lib/ladybug/energy_model/simulation_parameter.rb

```diff
@@ -0,0 +1,188 @@
+        if @hash[:simulation_control]
+          #Gets or creates new SimulationControl object from the OpenStudio model
+          openstudio_simulation_control = @openstudio_model.getSimulationControl
+          if @hash[:simulation_control][:do_zone_sizing]
+            openstudio_simulation_control.setDoZoneSizingCalculation(@hash[:simulation_control][:do_zone_sizing])
+          else 
+            openstudio_simulation_control.setDoZoneSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_zone_sizing][:default])
+          end
+          if @hash[:simulation_control][:do_system_sizing]
+            openstudio_simulation_control.setDoSystemSizingCalculation(@hash[:simulation_control][:do_system_sizing])
+          else
+            openstudio_simulation_control.setDoSystemSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_system_sizing][:default])
+          end
+          if @hash[:simulation_control][:do_plant_sizing]
+            openstudio_simulation_control.setDoPlantSizingCalculation(@hash[:simulation_control][:do_plant_sizing])
+          else
+            openstudio_simulation_control.setDoPlantSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_plant_sizing][:default])
+          end
+          if @hash[:simulation_control][:run_for_run_periods]
+            openstudio_simulation_control.setRunSimulationforWeatherFileRunPeriods(@hash[:simulation_control][:run_for_run_periods])
+          else
+            openstudio_simulation_control.setRunSimulationforWeatherFileRunPeriods(@@schema[:definitions][:SimulationControl][:properties][:run_for_run_periods][:default])
+          end
+          if @hash[:simulation_control][:run_for_sizing_periods]
+            openstudio_simulation_control.setRunSimulationforSizingPeriods(@hash[:simulation_control][:run_for_sizing_periods])
+          else
+            openstudio_simulation_control.setRunSimulationforSizingPeriods(@@schema[:definitions][:SimulationControl][:properties][:run_for_sizing_periods][:default])
+          end
+        end
+        if @hash[:shadow_calculation]
+          openstudio_shadow_calculation = @openstudio_model.getShadowCalculation
+          if @hash[:shadow_calculation][:calculation_frequency]
+            openstudio_shadow_calculation.setCalculationFrequency(@hash[:shadow_calculation][:calculation_frequency])
+          else
+            openstudio_shadow_calculation.setCalculationFrequency(@@schema[:definitions][:ShadowCalculation][:properties][:calculation_frequency][:default])
+          end
+          if @hash[:shadow_calculation][:maximum_figures]
+            openstudio_shadow_calculation.setMaximumFiguresInShadowOverlapCalculations(@hash[:shadow_calculation][:maximum_figures])
+          else
+            openstudio_shadow_calculation.setMaximumFiguresInShadowOverlapCalculations(@@schema[:definitions][:ShadowCalculation][:properties][:maximum_figures][:default])
+          end
```

Comment:
To map the `calculation_method` from the schema, the `setCalculationMethod` in the OpenStudio SDK is supported by v. 2.9 and is not available for the current OpenStudio version.

Reply:
I'm aware that this is a feature that Dan added before he left.  We are fine with only supporting 2.9 and above.

Reply:
Ah, wait, I didn't realize that we hadn't yet put the `calculation_method` in. Let's leave this unresolved until we can do that.